internal/pkg/cli/local_run.go

```diff
@@ -392,8 +431,13 @@ func (o *localRunOpts) runContainers(ctx context.Context, imageInfoList []clidep
 					LinePrefix: fmt.Sprintf("[%s] ", imageInfo.ContainerName),
 				},
 			}
-			if err := o.dockerEngine.Run(ctx, runOptions); err != nil {
-				return fmt.Errorf("run container: %w", err)
+			if err := o.dockerEngine.Run(context.Background(), runOptions); err != nil {
+				select {
+				case <-ctx.Done():
+					return nil
+				default:
+					return fmt.Errorf("run container: %w", err)
```

Comment:
you only need `ctx.Done()` if you want to _block_ until `Done()` returns. since you have `default` in this case, you don't actually want to block. this (and above) can probably be simplified to:
```suggestion
				if ctx.Err() != nil {
				  return nil
				}
				return fmt.Errorf("run container: %w", err)
```